File: app/src/main/java/com/wangwh/posterdemo/models/BackgroundViewAttrs.java
package com.wangwh.posterdemo.models;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.AppCompatImageView;

import com.bumptech.glide.Glide;

/**
 * @Author: 王文辉
 * @Date: 2022-04-19 14:36
 * @Desc: 背景属性
 * @Note:
 */
public class BackgroundViewAttrs extends ImageViewAttrs {

    @Override
    public int getW() {
        return ViewGroup.LayoutParams.MATCH_PARENT;
    }

    @Override
    public int getH() {
        return ViewGroup.LayoutParams.MATCH_PARENT;
    }

    @Override
    @NonNull
    public View createView(@NonNull Context context) {
        AppCompatImageView view = new AppCompatImageView(context);
        view.setScaleType(ImageView.ScaleType.FIT_XY);
        Glide.with(context).load(getUrl()).into(view);
        return view;
    }
}

File: app/src/main/java/com/wangwh/posterdemo/models/BackgroundViewAttrs.kt
package com.wangwh.posterdemo.models

import android.content.Context
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.widget.AppCompatImageView
import com.bumptech.glide.Glide

/**
 * @Author: 王文辉
 * @Date: 2022-04-19 10:45
 * @Desc: 定义海报背景图属性
 * @Note:
 */
class BackgroundViewAttrs(
    val url: String,
) : ViewAttrs(
    x = 0,
    y = 0,
    w = ViewGroup.LayoutParams.MATCH_PARENT,
    h = ViewGroup.LayoutParams.MATCH_PARENT
) {
    override fun createView(ctx: Context): View {
        val attrs = this
        return AppCompatImageView(ctx).apply {
            Glide.with(this).load(attrs.url).into(this)
        }
    }
}